exp/controllers/rosamachinepool_controller_test.go

```diff
@@ -60,3 +81,459 @@ func TestNodePoolToRosaMachinePoolSpec(t *testing.T) {
+			expect: func(m *mocks.MockOCMClientMockRecorder) {
+				m.GetNodePool(gomock.Any(), gomock.Any()).DoAndReturn(func(clusterId string, nodePoolID string) (*cmv1.NodePool, bool, error) {
+					nodePoolBuilder := nodePoolBuilder(rosaMachinePool(1).Spec, ownerMachinePool(1).Spec)
+					nodePool, err := nodePoolBuilder.ID("node-pool-1").Build()
+					g.Expect(err).To(BeNil())
+					return nodePool, true, nil
+				}).Times(1)
+				m.UpdateNodePool(gomock.Any(), gomock.Any()).DoAndReturn(func(clusterID string, nodePool *cmv1.NodePool) (*cmv1.NodePool, error) {
+					return nodePool, nil
+				}).Times(1)
+				m.CreateNodePool(gomock.Any(), gomock.Any()).Times(0)
+			},
+		},
+		{
+			name: "Nodepool is ready",
+			old:  rosaMachinePool(2),
+			new: &expinfrav1.ROSAMachinePool{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "rosa-machinepool",
+					Namespace: ns.Name,
+					UID:       "rosa-machinepool",
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ROSAMachinePool",
+					APIVersion: expinfrav1.GroupVersion.String(),
+				},
+				Spec: expinfrav1.RosaMachinePoolSpec{
+					NodePoolName: "test-nodepool",
+					Version:      "4.14.5",
+					Subnet:       "subnet-id",
+					InstanceType: "m5.large",
+				},
+				Status: expinfrav1.RosaMachinePoolStatus{
+					Ready:    true,
+					Replicas: 1,
+				},
+			},
+			result: ctrl.Result{},
+			expect: func(m *mocks.MockOCMClientMockRecorder) {
+				m.GetNodePool(gomock.Any(), gomock.Any()).DoAndReturn(func(clusterId string, nodePoolID string) (*cmv1.NodePool, bool, error) {
+					nodePoolBuilder := nodePoolBuilder(rosaMachinePool(2).Spec, ownerMachinePool(2).Spec)
+					statusBuilder := (&cmv1.NodePoolStatusBuilder{}).CurrentReplicas(1)
+					autoscalingBuilder := (&cmv1.NodePoolAutoscalingBuilder{}).MinReplica(1).MaxReplica(1)
+					nodePool, err := nodePoolBuilder.ID("node-pool-1").Autoscaling(autoscalingBuilder).Replicas(1).Status(statusBuilder).Build()
+					g.Expect(err).NotTo(HaveOccurred())
+
+					return nodePool, true, nil
+				}).Times(1)
+				m.UpdateNodePool(gomock.Any(), gomock.Any()).DoAndReturn(func(clusterID string, nodePool *cmv1.NodePool) (*cmv1.NodePool, error) {
+					statusBuilder := (&cmv1.NodePoolStatusBuilder{}).CurrentReplicas(1)
+					version := (&cmv1.VersionBuilder{}).RawID("4.14.5")
+					npBuilder := cmv1.NodePoolBuilder{}
+					updatedNodePool, err := npBuilder.Copy(nodePool).Status(statusBuilder).Version(version).Build()
+					g.Expect(err).NotTo(HaveOccurred())
+
+					return updatedNodePool, nil
+				}).Times(1)
+				m.CreateNodePool(gomock.Any(), gomock.Any()).Times(0)
+			},
+		},
+	}
+
+	createObject(g, secret, ns.Name)
+	createObject(g, identity, ns.Name)
+	defer cleanupObject(g, secret)
+	defer cleanupObject(g, identity)
+
+	for i, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// This is set by CAPI MachinePool reconcile
+			test.old.OwnerReferences = []metav1.OwnerReference{
+				{
+					Name:       ownerMachinePool(i).Name,
```

Comment:
Usually in k8s project we use the rand [lib](https://pkg.go.dev/k8s.io/apimachinery@v0.32.0/pkg/util/rand#String)  to generate a random string. If you can change it will be better otherwise add TODO note to change that.